### ISSUE TITLE
Feature enhancement: Create a Subsegments for operations handled by Powertools

### ISSUE THREAD

**Issue by mriccia** [aws-powertools/powertools-lambda-java#1505]
**Is your feature request related to a problem? Please describe.**

When using the idempotency module with Tracing: Active, it is not possible to see from the Trace how much time was spent inside Powertools code.
Adding a subsegment will make it easier to correlate the DynamoDB SDK calls, with the Idempotency functionality from Powertools.

**Describe the solution you'd like**

I would like an XRay subsegment to be created to show how long it's spent inside the Idempotency code.
This is an initial idea for discussion/consideration, it could be further expanded to other Powertools functionality.

**Comment by jeromevdl** [comment]
Good idea but the idempotency module works before and after the handler method is actually executed. Should we add 2 segments?
- One before when we check in the DB if the idempotency key is already there or insert it
- One after when we update the record

**Comment by jeromevdl** [comment]
Also, we may have a "dependency" issue. This should happen only if tracing is enabled / powertools-tracing is used. But we don't want to add a (maven) dependency to xray in the idempotency module...

**Comment by jeromevdl** [comment]
Let's do this in v2

**Comment by scottgerring** [comment]
Idempotency isn't the only module that calls AWS services. Do we want to add it to everything (e.g. the various params?)

> Also, we may have a "dependency" issue. This should happen only if tracing is enabled / powertools-tracing is used. But we don't want to add a (maven) dependency to xray / powertools-tracing in the idempotency module...

This will be the trick.

**Comment by scottgerring** [comment]
@heitorlessa wdyt?

**Comment by phipag** [comment]
This issue is linked to a discussion that we have at https://github.com/aws-powertools/powertools-lambda-java/discussions/1919. 

We are discussing of adding a tighter integration with Open Telemetry and if we should support customers in passing their OTEL tracer to Powertools utilities. This way, customers can also have traces of internal steps happening within the library. For example, in the case of batch processing this is really useful to understand the e2e picture of latency during batch processing.

**Comment by phipag** [comment]
Related OpenTelemetry RFC: https://github.com/aws-powertools/powertools-lambda-java/issues/1943
